pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnSingleValueReaderWriter.java

```diff
@@ -29,191 +29,108 @@
+ *   <li> Supports random reads and writes. </li>
+ * </ul>
+ */
 public class FixedByteSingleColumnSingleValueReaderWriter extends BaseSingleColumnSingleValueReaderWriter {
+  private static final Logger LOGGER = LoggerFactory.getLogger(FixedByteSingleColumnSingleValueReaderWriter.class);
 
-  public static Logger LOGGER = LoggerFactory.getLogger(FixedByteSingleColumnSingleValueReaderWriter.class);
-  WriterWithOffset currentWriter;
-  private List<WriterWithOffset> writers = new ArrayList<>();
-  private List<ReaderWithOffset> readers = new ArrayList<>();
-  private List<PinotDataBuffer> buffers = new ArrayList<>();
-  private final long chunkSizeInBytes;
-  private final int numRowsPerChunk;
-  private final int columnSizesInBytes;
-  private final RealtimeIndexOffHeapMemoryManager memoryManager;
-  private final String columnName;
-  private long numBytesInCurrentWriter = 0;
+  private List<WriterWithOffset> _writers = new ArrayList<>();
+  private List<ReaderWithOffset> _readers = new ArrayList<>();
+  private List<PinotDataBuffer> _dataBuffers = new ArrayList<>();
 
-  private static class WriterWithOffset implements Closeable {
-    final FixedByteSingleValueMultiColWriter writer;
-    final int startRowId;
-
-    private WriterWithOffset(FixedByteSingleValueMultiColWriter writer, int startRowId) {
-      this.writer = writer;
-      this.startRowId = startRowId;
-    }
-
-    @Override
-    public void close() {
-      writer.close();
-    }
-
-    public void setInt(int row, int value) {
-      writer.setInt(row - startRowId, 0, value);
-    }
-
-    public void setLong(int row, long value) {
-      writer.setLong(row - startRowId, 0, value);
-    }
+  private final long _chunkSizeInBytes;
+  private final int _numRowsPerChunk;
+  private final int _columnSizesInBytes;
 
-    public void setFloat(int row, float value) {
-      writer.setFloat(row - startRowId, 0, value);
-    }
-
-    public void setDouble(int row, double value) {
-      writer.setDouble(row - startRowId, 0, value);
-    }
-  }
-
-  private static class ReaderWithOffset implements Closeable {
-    final FixedByteSingleValueMultiColReader reader;
-    final int startRowId;
-
-    private ReaderWithOffset(FixedByteSingleValueMultiColReader reader, int startRowId) {
-      this.reader = reader;
-      this.startRowId = startRowId;
-    }
-
-    @Override
-    public void close() {
-      reader.close();
-    }
-
-    public int getInt(int row) {
-      return reader.getInt(row - startRowId, 0);
-    }
-
-    public long getLong(int row) {
-      return reader.getLong(row - startRowId, 0);
-    }
-
-    public float getFloat(int row) {
-      return reader.getFloat(row - startRowId, 0);
-    }
-
-    public double getDouble(int row) {
-      return reader.getDouble(row - startRowId, 0);
-    }
-
-    public FixedByteSingleValueMultiColReader getReader() {
-      return reader;
-    }
-  }
+  private final RealtimeIndexOffHeapMemoryManager _memoryManager;
```

Comment:
why are we hard coding realtimememory manager, this class should just take a memorymanager and not care about the implementation

Reply:
you can rename RealtimeIndexOffHeapMemoryManager to MemoryManager (or whatever generic name you like) and rename getSegmentName() method to getAllocationContext(), and you should be good to go.

The class is not correctly named, but that was intentionally so, since we were using it in the realtime context only.